homeassistant/components/tuya/__init__.py

```diff
@@ -50,17 +54,35 @@
 )
 
 
-def setup(hass, config):
+def setup(hass, config, retry_delay=FIRST_RETRY_TIME):
     """Set up Tuya Component."""
 
+    _LOGGER.info("Initializing %s domain.", DOMAIN)
+
     tuya = TuyaApi()
     username = config[DOMAIN][CONF_USERNAME]
     password = config[DOMAIN][CONF_PASSWORD]
     country_code = config[DOMAIN][CONF_COUNTRYCODE]
     platform = config[DOMAIN][CONF_PLATFORM]
 
+    try:
+        tuya.init(username, password, country_code, platform)
+    except NewConnectionError:
+        _LOGGER.warning(
+            "Connection error initializing %s domain. Will retry in %s seconds...",
+            DOMAIN,
```

Comment:
We don't need to specify domain. The logger is already specific for this domain.